sycl/test-e2e/Graph/RecordReplay/handlerless_enqueue_functions.cpp

```diff
@@ -0,0 +1,122 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+// Extra run to check for leaks in Level Zero using UR_L0_LEAKS_DEBUG
+// RUN: %if level_zero %{%{l0_leak_check} %{run} %t.out 2>&1 | FileCheck %s --implicit-check-not=LEAK %}
+
+// Test recording of several handlerless SYCL queue APIs (memset, memcpy,
+// parallel_for/nd_launch, single_task) inside a single function call
+// while graph recording is active. All operations are USM-based and occur via
+// eventless queue free functions or eventful queue shortcuts to bypass handler
+// path. Recording is performed over a non-inlined function call.
+
+#include "../graph_common.hpp"
+#include <cstdint>
+#include <cstring>
+#include <sycl/ext/oneapi/experimental/enqueue_functions.hpp>
+#include <sycl/properties/all_properties.hpp>
+#include <vector>
```

Comment:
This looks quite verbose to other related tests. Can you please check if includes are actually required, if so we might want to update `graph_common.hpp` ...

Reply:
The vector, stdint, and string includes are not necessary, so I removd them. `enqueue_functions.hpp` and `all_properties.hpp` are required here but not used across all tests, so I didn't add them to `graph_common.hpp`.

If we'd rather just have any needed includes in graph_common.hpp, then I can move those there.

Reply:
Thanks. No these two are fine, we should keep them extra as not all tests use properties or extensions like free enqueue functions. It will negatively impact compilation time if we just use a single combined header for all tests.